src/utils/legacyLayerParamUtils.js

```diff
@@ -173,3 +173,82 @@ export async function getKmlLayerFromLegacyAdminIdParam(adminId) {
         kmlMetadata,
     })
 }
+/**
+ * In the old mapviewer, it was possible to give a parameter in the following form :
+ * {layer.id}={features.id} example :
+ * https://map.geo.admin.ch/?ch.bav.haltestellen-oev=8577444,8505082,8507423,8577461 The goal was to
+ * allow people to create a link where some features were already selected, and the tooltip wouldn't
+ * be showing by default. In the current mapviewer, this is accomplished by giving the extra
+ * parameter `@features` on the layer
+ *
+ * @param {Query} params All parameters given to us through the query
+ * @param {Store} store The store
+ * @param {Query} newQuery The Query which will be sent to the current parameter parser
+ */
+export function handleLayerFeaturePreselection(params, store, newQuery) {
+    params.forEach((param_value, param_key) => {
+        if (store.state.layers.config.find((layer) => layer.id === param_key)) {
+            const featuresIds = param_value.split(',').join(':')
+            if (newQuery.layers?.includes(param_key)) {
+                newQuery.layers = createLayersParamForFeaturePreselection(
+                    param_key,
+                    param_value,
+                    newQuery.layers
+                )
+            } else if (newQuery.layers) {
+                newQuery.layers = `${newQuery.layers};${param_key}@features=${featuresIds}`
+            } else {
+                newQuery.layers = `${param_key}@features=${featuresIds}`
+            }
+        }
+    })
+}
+
+/**
+ * @param {String} layerId The layer Id for which we have features
+ * @param {String} featuresIds The features ids we need to add as a parameter. This a coma-separated
+ *   string
+ * @param {String} layers The new Query layers parameter, a semicolon separated string
+ * @returns
+ */
+function createLayersParamForFeaturePreselection(layerId, featuresIds, layers) {
+    const featuresArray = featuresIds.split(',')
+    // we go through each layer
+    const layersArray = layers.split(';')
+    for (let layerIndex = 0; layerIndex < layersArray.length; layerIndex++) {
+        if (layersArray[layerIndex].includes(layerId)) {
```

Comment:
you could `const` this `layersArray[layerIndex]` once and then reuse it bellow, might be easier to read